presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnValidator.java

```diff
@@ -16,82 +16,205 @@
-
-        return ImmutableList.of(new ColumnMatchResult<>(Objects.equals(controlChecksum, testChecksum), column, controlChecksum, testChecksum));
-    }
-
     public static Expression generateArrayChecksum(Expression column, Type type)
     {
         checkArgument(type instanceof ArrayType, "Expect ArrayType, found %s", type.getDisplayName());
         Type elementType = ((ArrayType) type).getElementType();
 
         if (elementType.isOrderable()) {
-            FunctionCall arraySort = new FunctionCall(QualifiedName.of("array_sort"), ImmutableList.of(column));
+            Expression arraySort = functionCall("array_sort", column);
 
             if (elementType instanceof ArrayType || elementType instanceof RowType) {
                 return new CoalesceExpression(
-                        new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(new TryExpression(arraySort))),
-                        new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(column)));
+                        functionCall("checksum", new TryExpression(arraySort)),
+                        functionCall("checksum", column));
             }
-            return new FunctionCall(QualifiedName.of("checksum"), ImmutableList.of(arraySort));
+            return functionCall("checksum", arraySort);
+        }
+        return functionCall("checksum", column);
+    }
+
+    private static Expression generateInfinityLambdaExpression(ArithmeticUnaryExpression.Sign sign)
+    {
+        ComparisonExpression lambdaBody = new ComparisonExpression(
+                ComparisonExpression.Operator.EQUAL,
+                new Identifier("x"),
+                new ArithmeticUnaryExpression(sign, functionCall("infinity")));
+        return new LambdaExpression(ImmutableList.of(new LambdaArgumentDeclaration(identifier("x"))), lambdaBody);
+    }
+
+    private static Expression generateLambdaExpression(String functionName)
+    {
+        return new LambdaExpression(
+                ImmutableList.of(new LambdaArgumentDeclaration(identifier("x"))),
+                functionCall(functionName, new Identifier("x")));
+    }
+
+    @Override
+    public List<ColumnMatchResult<ArrayColumnChecksum>> validate(Column column, ChecksumResult controlResult, ChecksumResult testResult)
+    {
+        checkArgument(
+                controlResult.getRowCount() == testResult.getRowCount(),
+                "Test row count (%s) does not match control row count (%s)",
+                testResult.getRowCount(),
+                controlResult.getRowCount());
+
+        boolean useFloatingPointPath = useFloatingPointPath(column);
+
+        ArrayColumnChecksum controlChecksum = toColumnChecksum(column, controlResult, useFloatingPointPath);
+        ArrayColumnChecksum testChecksum = toColumnChecksum(column, testResult, useFloatingPointPath);
+
+        // Not floating point case (we have '$checksum' column).
+        if (!useFloatingPointPath) {
```

Comment:
We don't need `useFloatingPointPath ` here, we can infer this information from ArrayColumnChecksum structure

Reply:
How so?

Reply:
if `ArrayColumnChecksum `'s checksum is not null, then it's regular ArrayColumnChecksum
If it is null, then it's the float point check

Reply:
> if `ArrayColumnChecksum `'s checksum is not null, then it's regular ArrayColumnChecksum If it is null, then it's the float point check

@kewang1024 

That was my 1st choice, but it all breaks on the corner case when we get checksums on empty tables (no rows) and then a bunch of checksum columns have nulls.

I realized after that if we are triggering floating point path checking the array element type - we should be consitent and always check using that approach. :)